coalib/results/TextRange.py

```diff
@@ -89,7 +89,35 @@ def end(self):
         return self._end
 
     def overlaps(self, other):
-        return self.start <= other.end and self.end >= other.start
+        first = False  # check for left boundary overlap
+        # self.start <= other.end
+
+        second = False  # check for right boundary overlap
+        # self.end >= other.start
+
+        if None not in (self.start.line, other.end.line):
+            if self.start.line == other.end.line:
+                if None in (self.start.column, other.end.column):
+                    first = True  # if column = None, consider whole line
+                else:
+                    first = self.start.column <= other.end.column
+            else:
+                first = self.start.line < other.end.line
+        else:  # If line = None, consider whole file
+            first = True
+
+        if None not in (self.end.line, other.start.line):
+            if self.end.line == other.start.line:
+                if None in (self.end.column, other.start.column):
+                    second = True
+                else:
+                    second = self.end.column >= other.start.column
+            else:
+                second = self.end.line > other.start.line
+        else:
+            second = True
```

Comment:
would It not make the `overlaps` method complicated ?
I have put comments to clarify `first` and `second`.